runners/core-java/src/test/java/org/apache/beam/runners/core/SplittableParDoTest.java

```diff
@@ -146,18 +151,23 @@ public void testBoundednessForUnboundedFn() {
         "Applying an unbounded SDF to a bounded collection produces a bounded collection",
         PCollection.IsBounded.UNBOUNDED,
         makeBoundedCollection(pipeline)
-            .apply("unbounded to bounded", new SplittableParDo<>(unboundedFn))
+            .apply("unbounded to bounded", new SplittableParDo.Bound<>(ParDo.of(unboundedFn)))
             .isBounded());
     assertEquals(
         "Applying an unbounded SDF to an unbounded collection produces an unbounded collection",
         PCollection.IsBounded.UNBOUNDED,
         makeUnboundedCollection(pipeline)
-            .apply("unbounded to unbounded", new SplittableParDo<>(unboundedFn))
+            .apply("unbounded to unbounded", new SplittableParDo.Bound<>(ParDo.of(unboundedFn)))
```

Comment:
Unfortunately that's not possible - there's a cyclic dependency between DoFnTester and ProcessFn: DoFnTester is constructed with ProcessFn, but ProcessFn needs an OutputWindowedValue that will refer to DoFnTester.

Well, it's possible, with some trickery, but it seems not worth it. I think setters are a fine way of wiring together objects with a cyclic dependency.